aten/src/ATen/TensorOptions.h

```diff
@@ -227,7 +221,7 @@ inline Tensor to(
     return tensor;
   }
   DeviceGuard guard(options.device());
-  return options.type().copy(tensor, non_blocking);
+  return at::getType(options).copy(tensor, non_blocking);
```

Comment:
can we do this in some other way?  One of the goals here was to be able to pull out the context from TensorOptions (so we can move TensorOptions to core without moving the Context).  I'm not really sure why this is a function on TensorOptions and not Tensor to be honest.

Reply:
Yeah, I agree this has no place on TensorOptions. @goldsborough, do you agree? This fix will be in another diff though :)

Reply:
It's probably here because TensorOptions.h is not available from Tensor.h. Which come to think of it, is totally the reverse of what you want.